tips-and-tricks/rolling_averages_with_dask.ipynb

```diff
@@ -0,0 +1,2047 @@
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1596153600000000000</th>\n",
+       "      <td>2020-07-31</td>\n",
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1596240000000000000</th>\n",
+       "      <td>2020-08-01</td>\n",
+       "      <td>NaN</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                          date  daily_avg_fare_amount\n",
+       "datenum                                              \n",
+       "1595462400000000000 2020-07-23                    NaN\n",
+       "1595548800000000000 2020-07-24                    NaN\n",
+       "1595635200000000000 2020-07-25                    9.0\n",
+       "1595721600000000000 2020-07-26                    NaN\n",
+       "1595808000000000000 2020-07-27                    NaN\n",
+       "1595894400000000000 2020-07-28                    NaN\n",
+       "1595980800000000000 2020-07-29                    NaN\n",
+       "1596067200000000000 2020-07-30                    NaN\n",
+       "1596153600000000000 2020-07-31                    NaN\n",
+       "1596240000000000000 2020-08-01                    NaN"
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "fares.tail(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "dask.dataframe.core.Series"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "type(fares)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our taxi dataset has some unlikely extreme dates on the outer edges, so I'm going to filter by date just to make sure we have reliable data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "t1 = datetime.datetime(2019, 1, 1,0,0,0)\n",
+    "t2 = np.datetime64(t1, 'ns')\n",
+    "t3 = t2.astype(int)\n",
+    "\n",
+    "t1 = datetime.datetime(2020, 8, 1,0,0,0)\n",
+    "t2 = np.datetime64(t1, 'ns')\n",
+    "t4 = t2.astype(int)\n",
+    "\n",
+    "taxi = taxi.loc[t3:t4]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Optional: Fill Missing Dates in Index\n",
+    "\n",
+    "If you don't have gaps in your date field, then you can ignore this entirely! But for many time series use cases, you'll have missing time points, and want to fill those in before calculating a rolling aggregation. The way we'll do that here is to create a brand new complete index, using pandas functions, and then we'll merge that with our starting dataset.\n",
```

Comment:
hard-won lessons from Uptake :')

Reply:
LOL yes I know the pain of time series. Do I ever!